### PR TITLE
#82 Separate user and household schedule

### PR DESCRIPTION
**Pull request by jurgen-potter**

### CODE REVIEW

File: adapters/incoming/rest/src/main/java/be/xplore/cookbook/rest/controller/household/HouseholdInviteController.java [hunk 1]
```diff
@@ -1,94 +1,94 @@
...
-            @PathVariable UUID inviteId
-    ) {
-        householdInviteService.revokeInvite(new RevokeInviteCommand(
-                new HouseholdInviteId(inviteId),
-                new HouseholdId(householdId),
-                new UserId(UUID.fromString(jwt.getSubject()))
-        ));
-    }
-}
+package be.xplore.cookbook.rest.controller.household;
```
Comment (Ftlbvd78) [inline adapters/incoming/rest/src/main/java/be/xplore/cookbook/rest/controller/household/HouseholdInviteController.java:1]: I would put this under householdInvite just like in the domain

File: adapters/incoming/rest/src/main/java/be/xplore/cookbook/rest/controller/schedule/HouseholdWeekScheduleController.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    @Transactional
+    @ResponseStatus(HttpStatus.CREATED)
+    public WeekScheduleDto createSchedule(
+            @AuthenticationPrincipal Jwt jwt,
+            @PathVariable UUID householdId,
+            @Valid @RequestBody CreateWeekScheduleDto dto
+    ) {
+        UserId userId = getUserIdFromJwt(jwt);
+        ScheduleOwner owner = ScheduleOwner.forHousehold(new HouseholdId(householdId));
+        return WeekScheduleDto.fromDomain(scheduleService.saveWeekSchedule(
```
Comment (Ftlbvd78) [inline adapters/incoming/rest/src/main/java/be/xplore/cookbook/rest/controller/schedule/HouseholdWeekScheduleController.java]: Split createWeekScheduleCommand into one for the user and one for the household, and change it here so that you don't need a ScheduleOwner

File: adapters/incoming/rest/src/main/java/be/xplore/cookbook/rest/controller/schedule/PersonalWeekScheduleController.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    @PostMapping
+    @Transactional
+    @ResponseStatus(HttpStatus.CREATED)
+    public WeekScheduleDto createSchedule(
+            @AuthenticationPrincipal Jwt jwt,
+            @Valid @RequestBody CreateWeekScheduleDto dto
+    ) {
+        UserId userId = getUserIdFromJwt(jwt);
+        ScheduleOwner owner = ScheduleOwner.forUser(userId);
```
Comment (Ftlbvd78) [inline adapters/incoming/rest/src/main/java/be/xplore/cookbook/rest/controller/schedule/PersonalWeekScheduleController.java]: Same comment here as in household week schedule

File: adapters/outgoing/jpa/src/main/java/be/xplore/cookbook/jpa/repository/recipe/RecipeRepositoryImpl.java [hunk 4]
```diff
@@ -87,11 +88,39 @@ public PagedResult<RecipeSummary> findAllSummariesWithFilter(
...
     @Override
-    public List<RecipeSummary> querySummaries(Paging paging, User user, String query) {
+    public List<RecipeSummary> findAllSummariesByUserIds(List<UserId> userIds) {
+        return jpaRecipeRepository.findByUser_IdInAndIsPublicTrue(userIds.stream().map(UserId::id).toList()).stream()
+                .map(JpaRecipeEntity::toSummary)
+                .toList();
+    }
+
+    @Override
+    public List<RecipeSummary> findAllPersonalSummariesByUserAndPreferences(UserPreferences preferences, User user) {
```
Comment (Ftlbvd78) [inline adapters/outgoing/jpa/src/main/java/be/xplore/cookbook/jpa/repository/recipe/RecipeRepositoryImpl.java]: Maybe call the findAllSummariesWithFilter method directly.

File: adapters/outgoing/jpa/src/main/java/be/xplore/cookbook/jpa/repository/weekschedule/JpaWeekScheduleRepository.java [hunk 5]
```diff
@@ -9,10 +10,24 @@
...
+            UUID ownerId,
+            ScheduleOwnerType ownerType
+    );
 
-    List<JpaWeekScheduleEntity> findByUserIdAndWeekStartDateBetweenOrderByWeekStartDateDesc(
-            UUID userId, LocalDate startFrom, LocalDate startTo);
+    List<JpaWeekScheduleEntity>
+    findByOwnerOwnerIdAndOwnerOwnerTypeAndWeekStartDateBetweenOrderByWeekStartDateDesc(
+            UUID ownerId,
+            ScheduleOwnerType ownerType,
```
Comment (Ftlbvd78) [inline adapters/outgoing/jpa/src/main/java/be/xplore/cookbook/jpa/repository/weekschedule/JpaWeekScheduleRepository.java]: Is ownertype really needed in this method and in the method below? I suppose that if you know the ownerId that should be enough, together with weekStartDate, to resolve a week schedule.

File: application/src/test/java/be/xplore/cookbook/rest/controller/recipe/recipeDiscoveryController/SearchHouseholdRecipeSummariesTests.java [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+    private static final String RECIPE_NAME = "Test Recipe";
+
+    @Override
+    protected String[] getTablesToClear() {
+        return new String[]{"recipe_ingredients", "recipe_steps", "recipes", "ingredients", "users", "households",
+                "households_members"};
+    }
+
+    @Test
+    void searchHouseholdRecipeSummaries_shouldReturnPublicRecipesOfMembers_whenQueryMatches() throws Exception {
```
Comment (Ftlbvd78) [inline application/src/test/java/be/xplore/cookbook/rest/controller/recipe/recipeDiscoveryController/SearchHouseholdRecipeSummariesTests.java]: Maybe also add a test where the creator searches household recipe summaries. This isn't that much of a change, but if something were to change in the queries, this test would help cover that edge case.

**Review by Ftlbvd78 — CHANGES_REQUESTED** [review]
Also, please fix the directory structure in the tests. You now sometimes have ingredient.ingredientController, and other times it is just household. Please remove the extra directory since I don't find it providing any meaningful addition. Also, don't forget to remove the random DB file.

**Review by Ftlbvd78 — DISMISSED** [review]

**Review by Ftlbvd78 — APPROVED** [review]
